Parsers/ASimDns/Parsers/ASimDnsInfobloxNIOS.yaml

```diff
@@ -26,49 +26,47 @@ ParserQuery: |
-    | extend 
-        DnsResponseCodeName = tostring(extract[0]),
-        DnsFlags = tostring(extract[1]),
-        DnsResponseName = tostring(extract[2])
-    | extend 
-        EventSubType = "response"
-    | project-away extract, rest_of_string,SyslogMessage, ProcessName, ProcessID, Facility, SeverityLevel, HostName
+    | parse SyslogMessage with *
+        "client " SrcIpAddr: string
+        "#" SrcPortNumber: string
+        " " NetworkProtocol: string
+        ": query: " DnsQuery: string
+        " " DnsQueryClassName: string
+        " " DnsQueryTypeName: string
+        " response: " DnsResponseCodeName: string
+        " " DnsFlags: string
+    | extend DNSResourceRecordIndex= indexof(DnsFlags, " ")
+    | extend DNSResourceRecord =iif(DNSResourceRecordIndex != "-1", substring(DnsFlags, DNSResourceRecordIndex), "")
+    | extend DnsFlags =iif(DNSResourceRecordIndex != "-1", substring(DnsFlags, 0, DNSResourceRecordIndex), DnsFlags)
+    | extend EventSubType = "response"
+    | project-away DNSResourceRecordIndex,SyslogMessage, ProcessName, ProcessID, Facility, SeverityLevel, HostName
   };
   let request = (disabled: boolean=false) {
     Syslog 
     | where not(disabled)
     | where ProcessName == "named" and SyslogMessage has_all ("client", "query:") and SyslogMessage !has "response:"
-    // -- trim_start needed to remove optional @xxxx identifier.
-    | extend SyslogMessage = trim_start (@'^(?:[\w\: ]+)?client (?:@.*? )?', SyslogMessage)
-    // -- trim_end needed to remove optional repetative IP address at the end.
-    | extend SyslogMessage = trim_end (@' \([\d\.]+\)', SyslogMessage)
-    | extend SyslogMessage = replace_string (SyslogMessage, '\\ ', '@@@')
-    | parse SyslogMessage with
-        SrcIpAddr:string "#" 
-        SrcPortNumber:int *
-        "query: "
-        DnsQuery:string " "
-        DnsQueryClassName:string " "
-        DnsQueryTypeName:string " "
-        DnsFlags:string
+    | parse SyslogMessage with *
+        "client " restData: string
+    | extend restData = iif(restData startswith "@", (substring(restData, indexof(restData, " "))), restData)
+    | extend restData = replace_string(restData,"\\ ","@@@")
+    | parse restData with 
+        SrcIpAddr: string
+        "#" SrcPortNumber: string
+        " " *
+        "query: " DnsQuery: string
+        " " DnsQueryClassName: string
+        " " DnsQueryTypeName: string
+        " " DnsFlags: string
     | extend DnsQuery = replace_string (DnsQuery, '@@@', ' ')
+    | extend ServerIPAddressIndex= indexof(DnsFlags, " ")
+    | extend ServerIPAddress = iif(ServerIPAddressIndex != "-1", substring(DnsFlags, ServerIPAddressIndex), "")
+    | extend ServerIPAddress = replace_regex(ServerIPAddress,@"[()]","")
+    | extend DnsFlags =iif(ServerIPAddressIndex != "-1", substring(DnsFlags, 0, ServerIPAddressIndex), DnsFlags)
+    | extend SrcPortNumber = replace_regex(SrcPortNumber,@"[^\d]","")
```

Comment:
Not sure I understand this line. First SrcPortNumber has to be an integer, which is how it is extracted in the original. and then, why would you remove the first character?

Reply:
In few logs, I found 'SrcPortNumber' ending with a ':' This was breaking the 'parse' logic as ':' was not getting saved in an 'int' type. sample logs:

1. query: info: client 1.2.3.4#60010: view standard: query: blip.prefetch.net IN AAAA +
2. client 1.2.3.4#42872 (prefetch.net): query: prefetch.net IN ANY + (1.2.3.4)

Previously I was using regex to remove ':' in 'SrcPortNumber'. Now I changed the logic to use replace_string and then convert to 'int' type. This would take care of intermittent ":" beside port numbers.

[ extend SrcPortNumber = toint(replace_string(SrcPortNumber,":","")) ]

Reply:
wouldn't using 

  **SrcPortNumber:int * " "** 

solve the issue?

int would ensure that the colon is not captures, and the * will capture it.

Reply:
It works. Thanks Ofer. Now using:
"#" SrcPortNumber: int *